selfservice/strategy/passkey/passkey_registration.go

```diff
@@ -0,0 +1,358 @@
+		return s.handleRegistrationError(w, r, regFlow, params, err)
+	}
+
+	if len(params.Register) == 0 {
+		return s.createPasskey(r, w, regFlow, params)
+	}
+
+	if err := flow.MethodEnabledAndAllowed(ctx, regFlow.GetFlowName(), params.Method, params.Method, s.d); err != nil {
+		return s.handleRegistrationError(w, r, regFlow, params, err)
+	}
+
+	if len(params.Traits) == 0 {
+		params.Traits = json.RawMessage("{}")
+	}
+	ident.Traits = identity.Traits(params.Traits)
+
+	webAuthnSession := gjson.GetBytes(regFlow.InternalContext, flow.PrefixInternalContextKey(s.ID(), InternalContextKeySessionData))
+	if !webAuthnSession.IsObject() {
+		return s.handleRegistrationError(w, r, regFlow, params, errors.WithStack(
+			herodot.ErrInternalServerError.WithReasonf("Expected WebAuthN in internal context to be an object.")))
+	}
+
+	var webAuthnSess webauthn.SessionData
+	if err := json.Unmarshal([]byte(webAuthnSession.Raw), &webAuthnSess); err != nil {
+		return s.handleRegistrationError(w, r, regFlow, params, errors.WithStack(
+			herodot.ErrInternalServerError.WithReasonf("Expected WebAuthN in internal context to be an object but got: %s", err)))
+	}
+
+	webAuthnResponse, err := protocol.ParseCredentialCreationResponseBody(strings.NewReader(params.Register))
+	if err != nil {
+		return s.handleRegistrationError(w, r, regFlow, params, errors.WithStack(
+			herodot.ErrBadRequest.WithReasonf("Unable to parse WebAuthn response: %s", err)))
+	}
+
+	webAuthn, err := webauthn.New(s.d.Config().PasskeyConfig(ctx))
+	if err != nil {
+		return s.handleRegistrationError(w, r, regFlow, params, errors.WithStack(
+			herodot.ErrInternalServerError.WithReasonf("Unable to get webAuthn config.").WithDebug(err.Error())))
+	}
+
+	credential, err := webAuthn.CreateCredential(webauthnx.NewUser(webAuthnSess.UserID, nil, webAuthn.Config), webAuthnSess, webAuthnResponse)
+	if err != nil {
+		if devErr := new(protocol.Error); errors.As(err, &devErr) {
+			s.d.Logger().WithError(err).WithField("error_devinfo", devErr.DevInfo).Error("Failed to create WebAuthn credential")
+		}
+		return s.handleRegistrationError(w, r, regFlow, params, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Unable to create WebAuthn credential: %s", err)))
+	}
+
+	credentialWebAuthn := identity.CredentialFromWebAuthn(credential, true)
+	credentialsConfig, err := json.Marshal(identity.CredentialsWebAuthnConfig{
+		Credentials: identity.CredentialsWebAuthn{*credentialWebAuthn},
+		UserHandle:  webAuthnSess.UserID,
+	})
+	if err != nil {
+		return s.handleRegistrationError(w, r, regFlow, params, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Unable to encode identity credentials.").WithDebug(err.Error())))
+	}
+
+	ident.UpsertCredentialsConfig(s.ID(), credentialsConfig, 1)
+	passkeyCred, _ := ident.GetCredentials(s.ID())
+	passkeyCred.Identifiers = []string{string(webAuthnSess.UserID)}
```

Comment:
I think we should  return an easy to understand error if `webAuthnSess.UserID` is empty. If it is empty, the current validation will pass it as valid

```
	c := i.GetCredentialsOr(identity.CredentialsTypePasskey, &identity.Credentials{})
	if len(c.Identifiers) == 0 {
		return schema.NewMissingIdentifierError()
	}
```

and will lead to an empty string identifier. We had this problem once before in the WebAuthn code.

Reply:
In this case, `webAuthnSess.UserID` is not user-supplied. It comes from the internal context, which is set here: https://github.com/ory/kratos/blob/60a1817c6660d1eb668a9994026da385416dcf44/selfservice/strategy/passkey/passkey_registration.go#L295-L303

I can add another check to make sure it really is not empty, but it would be an internal server error, as the user can't do anything against it.